scipy/spatial/transform/_rigid_transformation.pyx

```diff
@@ -0,0 +1,1978 @@
+                [0., 0., 0., 1.]],
+               [[1., 0., 0., 0.],
+                [0., 1., 0., 0.],
+                [0., 0., 1., 0.],
+                [0., 0., 0., 1.]]])
+        >>> tf.single
+        False
+        >>> len(tf)
+        2
+        """
+        if num is None:  # single
+            return cls(np.eye(4, dtype=float), normalize=False, copy=False)
+        else:
+            return cls(np.eye(4, dtype=float)[None, :, :].repeat(num, axis=0),
+                       normalize=False, copy=False)
+
+    @cython.embedsignature(True)
+    @classmethod
+    @_transition_to_rng('random_state', position_num=2)
+    def random(cls, num=None, rng=None):
+        """Initialize a random transformation.
+
+        This results in a random rotation combined with a random translation
+        vector, randomly sampled from the unit sphere.
+
+        Parameters
+        ----------
+        num : int, optional
+            Number of random transformations to generate. If None (default),
+            then a single transformation is generated.
+        rng : `numpy.random.Generator`, optional
+            Pseudorandom number generator state. When `rng` is None, a new
+            `numpy.random.Generator` is created using entropy from the
+            operating system. Types other than `numpy.random.Generator` are
+            passed to `numpy.random.default_rng` to instantiate a `Generator`.
+
+        Returns
+        -------
+        transformation : `RigidTransformation` instance
+            A single transformation or a stack of transformations.
+
+        Examples
+        --------
+        >>> from scipy.spatial.transform import RigidTransformation as Tf
+        >>> import numpy as np
+        >>> rng = np.random.default_rng(seed=123)
+        >>> tf = Tf.random(rng=rng)
+        >>> tf.as_matrix()
+        array([[-0.27687894,  0.08111092, -0.95747536,  0.21419513],
+               [ 0.43673235, -0.87694466, -0.20058143,  0.19815685],
+               [-0.85592225, -0.47369724,  0.20738374, -0.95648017],
+               [ 0.        ,  0.        ,  0.        ,  1.        ]])
+        >>> tf.single
+        True
+
+        The translation component will be a random unit vector:
+
+        >>> np.linalg.norm(tf.translation)
+        1.0
+
+        Multiple transformations can be generated at once:
+
+        >>> tf = Tf.random(3)
+        >>> tf.single
+        False
+        >>> len(tf)
+        3
+        """
+        rng = check_random_state(rng)
+        r = Rotation.random(num, rng=rng)
+        t = Rotation.random(num, rng=rng).apply([1, 0, 0])
+        return cls.from_components(t, r)
+
+    @cython.embedsignature(True)
+    @classmethod
+    def concatenate(cls, transformations):
+        """
+        Concatenate a sequence of `RigidTransformation` objects into a
+        single object.
+
+        Parameters
+        ----------
+        transformations : sequence of `RigidTransformation` objects, or
```

Comment:
Perhaps better to simply write "sequence of `RigidTransformation`" and on the next line clarify that both variants are supported.